lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -2140,10 +2306,17 @@ func (d *zfs) readonlySnapshot(vol Volume) (string, revert.Hook, error) {
 func (d *zfs) BackupVolume(vol Volume, tarWriter *instancewriter.InstanceTarWriter, optimized bool, snapshots []string, op *operations.Operation) error {
 	// Handle the non-optimized tarballs through the generic packer.
 	if !optimized {
+		dataset := d.dataset(vol, false)
+
+		volumeType, err := d.getDatasetProperty(dataset, "type")
+		if err != nil {
+			return err
+		}
+
 		// Because the generic backup method will not take a consistent backup if files are being modified
 		// as they are copied to the tarball, as ZFS allows us to take a quick snapshot without impacting
 		// the parent volume we do so here to ensure the backup taken is consistent.
-		if vol.contentType == ContentTypeFS {
+		if vol.contentType == ContentTypeFS && volumeType == "filesystem" {
```

Comment:
Why can't we use `!d.isBlockBacking(vol)` here and avoid the call to `getDatasetProperty`?

Reply:
Although I think we should still take a snapshot and mount it for `genericVFSBackupVolume` to read from.
Also if I'm reading this correctly, you're expecting `genericVFSBackupVolume` to produce a generirc (non-optimized) block volume backup.

But that raises an interesting dilemma. For all other storage pools a generic export should be importable into other storage pools of a different driver. 

So I think really the block volume should be mounted and the custom mount path passed to `genericVFSBackupVolume` so it creates a true filesystem generic backup.

Then the ZFS specific import function needs to be updated (if not already) to understand that if the volume's config indicates it was formerly a ZFS block volume, that the files in the backup are unpacked into a fresh block volume filesystem of the same type.

Reply:
A test for this is whether you can import an export produced here into a btrfs or dir pool and the container still start.